addons/website_event_track/static/src/js/website_event_track.js

```diff
@@ -96,3 +96,16 @@ publicWidget.registry.websiteEventTrack = publicWidget.Widget.extend({
         }
     },
 });
+
+publicWidget.registry.eventTrackForm = publicWidget.Widget.extend({
+    selector: '.o_wevent_event_search_box_track',
+    events: {
+        'search .search-query': '_onClearSearchInput',
+    },
+
+    _onClearSearchInput: function () {
```

Comment:
Here, you can then check if the input value is an empty string and if so, submit the form.
I don't think redefining the action is necessary :thinking:  To check